bin/git-forgit

```diff
@@ -339,6 +339,77 @@ _forgit_diff() {
     return $fzf_exit_code
 }
 
+_forgit_exec_show() {
+    _forgit_show_git_opts=()
+    _forgit_parse_array _forgit_show_git_opts "$FORGIT_SHOW_GIT_OPTS"
+    git show --pretty="" --diff-merges=first-parent --color=always "${_forgit_show_git_opts[@]}" "$@"
+}
+
+_forgit_show_view() {
+    local input_line=$1
+    local diff_context=$2
+    local commit=$3
+    local repo
+    repo=$(git rev-parse --show-toplevel)
+    cd "$repo" || return 1
+    echo "$input_line" | _forgit_get_files_from_diff_line | xargs -0 \
+        "$FORGIT" exec_show "${commit}^{commit}" -U"$diff_context" -- | _forgit_pager diff
+}
+
+_forgit_show_enter() {
+    file=$1
+    commit=$2
+    _forgit_show_view "$file" "$_forgit_fullscreen_context" "${commit}"
+}
+
+# git show viewer
+_forgit_show() {
+    _forgit_inside_work_tree || return 1
+    local files opts commit escaped_commit
+    files=()
+    if [[ $# -ne 0 ]]; then
+        if git rev-parse "$1" -- &>/dev/null ; then
+            commit="$1" && files=("${@:2}")
+        else
+            commit="HEAD" && files=("$@")
+        fi
+    else
+        commit="HEAD"
+    fi
+    # Escape opening brackets to support stashes (see comment in _forgit_diff)
+    escaped_commit=${commit//\{/\\\\\{}
+    opts="
+        $FORGIT_FZF_DEFAULT_OPTS
+        +m -0 --bind=\"enter:execute($FORGIT show_enter {} $escaped_commit | $FORGIT pager enter)\"
+        --preview=\"$FORGIT show_view {} '$_forgit_preview_context' $escaped_commit\"
+        --bind=\"alt-e:execute($FORGIT edit_diffed_file {})+refresh-preview\"
+        --bind 'ctrl-t:transform:[[ ! \$FZF_PREVIEW_LABEL =~ Diff ]] &&
+                echo \"change-preview-label( Diff )+refresh-preview\" ||
+                echo \"change-preview-label( Commit Message )+refresh-preview\"'
+        --preview '[[ \$FZF_PREVIEW_LABEL =~ Diff ]] &&
+                $FORGIT show_view {} '$_forgit_preview_context' $escaped_commit ||
+                git show --quiet --color=always \${FZF_PROMPT%% *}'
```

Comment:
I don't like that this introduces more deferred execution again. In this case we'd actually also have to do weird escaped quoting:

```suggestion
        --bind 'ctrl-t:transform:[[ ! \"\$FZF_PREVIEW_LABEL\" =~ Diff ]] &&
                echo \"change-preview-label( Diff )+refresh-preview\" ||
                echo \"change-preview-label( Commit Message )+refresh-preview\"'
        --preview '[[ \"\$FZF_PREVIEW_LABEL\" =~ Diff ]] &&
                $FORGIT show_view {} '$_forgit_preview_context' $escaped_commit ||
                git show --quiet --color=always \"\${FZF_PROMPT%% *}\"'
```

How about we move the part for the preview into `_forgit_show_view` or create a separate function for it? I think this would make it easier to maintain.

Reply:
Moving the preview part to a new function was a very good idea, thanks for that! Implemented that in the new version. I left the `--bind` argument as deferred code, though, as this is exactly how `fzf` shows it in their examples and the echoed strings would not make sense without the context of `--bind`. Do you agree?

Reply:
Yes, I think there is now way to move the `--bind` arguments out of deferred code, which is fine for me. We should probably double quote `$FZF_PREVIEW_LABEL` in the deferred code.

Reply:
I added the double quotes like you suggested. Looks a bit ugly, but works. I wanted to keep the double quotes for the complete `--bind` argument to be consistent with all the other args, so I had to double escape the quotes now.